src/apprt/gtk/Surface.zig

```diff
@@ -1264,6 +1265,90 @@ pub fn showDesktopNotification(
+                .system => system: {
+                    // FIXME: when surface is converted to zig-gobject
+                    const native = gtk.Widget.getNative(@ptrCast(@alignCast(self.overlay))) orelse break :system;
+                    const surface = native.getSurface() orelse break :system;
+                    surface.beep();
+                },
+                // Play a user-specified audio file.
+                .audio => audio: {
+                    var arena = std.heap.ArenaAllocator.init(self.app.core_app.alloc);
+                    defer arena.deinit();
+                    const alloc = arena.allocator();
+
+                    const filename = self.app.config.@"bell-audio" orelse break :audio;
+                    const pathname = pathname: {
+                        if (std.fs.path.isAbsolute(filename))
+                            break :pathname alloc.dupeZ(u8, filename) catch |err| {
+                                log.warn("unable to allocate space for bell audio pathname: {}", .{err});
+                                break :audio;
+                            }
+                        else
+                            break :pathname std.fs.path.joinZ(alloc, &.{
+                                internal_os.xdg.config(alloc, .{ .subdir = "ghostty/media" }) catch |err| {
+                                    log.warn("unable to determine media config subdir: {}", .{err});
+                                    break :audio;
+                                },
+                                filename,
+                            }) catch |err| {
+                                log.warn("unable to allocate space for bell audio pathname: {}", .{err});
+                                break :audio;
+                            };
+                    };
+
+                    const media_file = gtk.MediaFile.newForFilename(pathname);
+                    const media_stream = media_file.as(gtk.MediaStream);
+
+                    _ = gobject.Object.signals.notify.connect(
+                        media_stream,
+                        ?*anyopaque,
+                        gtkStreamError,
+                        null,
+                        .{ .detail = "error" },
+                    );
+                    _ = gobject.Object.signals.notify.connect(
+                        media_stream,
+                        ?*anyopaque,
+                        gtkStreamEnded,
+                        null,
+                        .{ .detail = "ended" },
+                    );
+
+                    media_stream.setVolume(1.0);
+                    media_stream.play();
+                },
+            }
+        }
+    }
+}
+
+/// Handle a stream that is in an error state.
+fn gtkStreamError(media_stream: *gtk.MediaStream, _: *gobject.ParamSpec, _: ?*anyopaque) callconv(.C) void {
+    const media_file = gobject.ext.cast(gtk.MediaFile, media_stream) orelse return;
```

Comment:
FYI you can avoid the cast by just passing in the media file directly above without `.as(gtk.MediaStream)`, then the first parameter would just be a `gtk.MediaFile`

Reply:
Actually, you can't (I tried). Since the property definition is on `gtk.MediaStream` the signal is expecting a function with a `*gtk.MediaStream` as the first parameter.

Reply:
That doesn't make much sense... child classes inherit all properties from their parent class. For example, you can still listen for changes of the `GtkWindow:decorated` property on an AdwApplicationWindow since it subclasses GtkWindow

Reply:
Well, I coulda _swore_ that this broke earlier, but ¯\_(ツ)_/¯. Fixed now.